File: Day1/one.py
l1: list[int] = []
l2: list[int] = []

while True:
    try:
        line = input()
        [x, y] = list(map(int, line.split()))
        l1.append(x)
        l2.append(y)
    except:
        break


l1.sort()
l2.sort()


# TODO: Type the lambda
# fn = lambda x : lambda y : abs(x-y)
def fn(xy: tuple[int, int]):
    [x, y] = xy
    return abs(x - y)


print(sum(map(fn, zip(l1, l2))))

File: Day1/one.hs
--
--    author:  bernborgess
--    problem: one - AoC2024
--    created: 02.December.2024 08:20:15
--

import Data.List (sort, transpose)

solve :: String -> String
solve input =
  let ws = map (map read . words) (lines input)
      [l1, l2] = map sort $ transpose ws
      ans = sum $ map abs $ zipWith (-) l1 l2
   in show ans

main :: IO ()
main = interact solve
